service/serialize.go

```diff
@@ -0,0 +1,153 @@
+package service
+
+import (
+	"strconv"
+
+	"github.com/mesg-foundation/engine/hash/hashserializer"
+)
+
+// HashSerialize returns the hashserialized string of this type
+func (data *Service) HashSerialize() string {
+	if data == nil {
+		return ""
+	}
+	ser := hashserializer.New()
+	ser.AddString("1", data.Name)
+	ser.AddString("2", data.Description)
+	ser.Add("5", serviceTasks(data.Tasks))
+	ser.Add("6", serviceEvents(data.Events))
+	ser.Add("7", serviceDependencies(data.Dependencies))
+	ser.Add("8", data.Configuration)
+	ser.AddString("9", data.Repository)
+	ser.AddString("12", data.Sid)
+	ser.AddString("13", data.Source)
+	return ser.HashSerialize()
+}
+
+// HashSerialize returns the hashserialized string of this type
+func (data Service_Configuration) HashSerialize() string {
+	ser := hashserializer.New()
+	ser.AddStringSlice("1", data.Volumes)
+	ser.AddStringSlice("2", data.VolumesFrom)
+	ser.AddStringSlice("3", data.Ports)
+	ser.AddStringSlice("4", data.Args)
```

Comment:
Not matching the current proto but I think the current proto is wrong
https://github.com/mesg-foundation/engine/blob/6c563e098f107e1fb063252aa5a3f339ab66abd3/protobuf/types/service.proto#L122-L124

Reply:
i fix the proto